legal-api/src/legal_api/models/action.py

```diff
@@ -0,0 +1,43 @@
+# Copyright © 2025 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module holds data for actions."""
+
+from sqlalchemy import func
+
+from .db import db
+
+
+class Action(db.Model):
+    """This class manages all of the actions."""
+
+    __tablename__ = 'actions'
+
+    id = db.Column(db.Integer, primary_key=True)
+    action_name = db.Column('action_name', db.String(100), nullable=False, unique=True)
+    description = db.Column('description', db.String(255), nullable=True)
+    created_date = db.Column('created_date', db.DateTime(timezone=True), default=func.now())
+    last_modified = db.Column('last_modified', db.DateTime(timezone=True), onupdate=func.now(), default=func.now())
+    created_by_id = db.Column(db.Integer, nullable=True)
+    modified_by_id = db.Column(db.Integer, nullable=True)
+
+    authorized_roles = db.relationship(
+        'AuthorizedRoleAction',
+        back_populates='action',
+        cascade='all, delete-orphan'
+    )
+
+    def save(self):
```

Comment:
possible base class for these?

Reply:
Could probably do the same thing for `created_date, last_modified, created_by_id, modified_by_id`

Reply:
https://github.com/bcgov/sbc-auth/blob/main/auth-api/src/auth_api/models/base_model.py
https://github.com/bcgov/sbc-pay/blob/main/pay-api/src/pay_api/models/audit.py#L26
https://github.com/bcgov/sbc-pay/blob/main/pay-api/src/pay_api/models/base_model.py